aptos-move/block-executor/src/task.rs

```diff
@@ -33,6 +33,7 @@ pub enum ExecutionStatus<O, E> {
     /// Transaction was executed successfully, but will skip the execution of the trailing
     /// transactions in the list
     SkipRest(O),
+    MaterializedSkipRest(O),
```

Comment:
nit: it might be better to add a dynamic flag to SkipRest. maybe DirectMaterialized: bool, or smt similar, that could supply more context. Depending on how deep we want to go, but might even make sense for this indicator to be a part of Output.

can give slightly more unified handling (lots of handling is the same with SkipRest) and test coverage (we have tests that create SkipRest, and it'd also be easier to play with the new flag). I think if "direct/materialized vs normal" were to be a property of the output, then it'd be a way more clear-cut decision: wouldn't even have to change the types, and would also technically be allowed to consider the property in other contexts. If it doesn't make as much sense, then I'm okay with both.